site/app/controllers/admin/ReportController.php

```diff
@@ -749,15 +895,14 @@ public function autoRainbowGradesStatus() {
         }
 
         $debug_contents = file_get_contents($debug_output_path);
+        $debug_contents = trim($debug_contents);
+        $was_successful = strrpos($debug_contents, 'Done') === (strlen($debug_contents) - strlen('Done'));
```

Comment:
Would [`str_ends_with()`](https://www.php.net/manual/en/function.str-ends-with.php) be helpful here?

Reply:
This is a good point, I was unaware of it, looks like it is php8 thing.